bin/alpen-reth/src/main.rs

```diff
@@ -83,11 +95,84 @@ fn main() {
             });
 
             let handle = node_builder.launch().await?;
+
+            // Create and add the custom RLPx subprotocol
+            // See: crates/reth/node/src/head_gossip/
+            let (gossip_tx, mut gossip_rx) = mpsc::unbounded_channel();
+            let handler = HeadGossipProtocolHandler {
+                state: HeadGossipState { events: gossip_tx },
+            };
+            handle
+                .node
+                .network
+                .add_rlpx_sub_protocol(handler.into_rlpx_sub_protocol());
+
+            // This task will handle gossip events
+            let mut state_events = handle.node.provider.subscribe_to_canonical_state();
+            tokio::spawn(async move {
+                let mut connections: HashMap<PeerId, mpsc::UnboundedSender<HeadGossipCommand>> =
+                    HashMap::new();
+
+                loop {
+                    tokio::select! {
+                        Some(event) = gossip_rx.recv() => {
+                            match event {
```

Comment:
I don't think this should be in `main`, can we not split it out into a separate (service?) crate so that it could be reusable or at least modularized?